src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Performance/PreferIsEmptyOverCountTests.cs

```diff
@@ -0,0 +1,360 @@
+        End Get
+        Set(ByVal value As ConcurrentDictionary(Of String, String))
+
+            If value Is Nothing OrElse value.Count = 0 Then
+                Throw New ArgumentException(NameOf(value))
+            End If
+
+            _myDictionary = value
+        End Set
+    End Property
+End Class
+";
+
+            string vbFix = @"
+Imports System
+Imports System.Collections.Concurrent
+
+Public Class Test
+    Private _myDictionary As ConcurrentDictionary(Of String, String)
+
+    Public Property MyDictionary As ConcurrentDictionary(Of String, String)
+        Get
+            Return _myDictionary
+        End Get
+        Set(ByVal value As ConcurrentDictionary(Of String, String))
+
+            If value Is Nothing OrElse value.IsEmpty Then
+                Throw New ArgumentException(NameOf(value))
+            End If
+
+            _myDictionary = value
+        End Set
+    End Property
+End Class
+";
+            await VerifyVB.VerifyCodeFixAsync(
+                vbInput,
+                VerifyVB.Diagnostic(UseCountProperlyAnalyzer.s_rule_CA1836).WithSpan(14, 40, 14, 55),
+                vbFix);
+        }
+
+//        [Fact]
+//        public Task QuickTest()
+//        {
+//            return VerifyCS.VerifyCodeFixAsync(
+//@"using System;
+
+//public class Test
+//{
+//    private int Count {get;}
+//    public int Length {get;}
+//    //public bool IsEmpty;
+//    public bool Foo => Count > 0;
+//}",
+//                    VerifyCS.Diagnostic(UseCountProperlyAnalyzer.s_rule_CA1836),
+//@"using System;
+
+//public class Test
+//{
+//    private static int Count = 0;
+//    public bool Foo => !GetData().IsEmpty;
+//}");
+//        }
+
+        [Theory]
+        [InlineData("(Count) > 0")]
+        [InlineData("Count > (0)")]
+        [InlineData("(Count) > (0)")]
+        [InlineData("(this.Count) > 0")]
+        [InlineData("this.Count > (0)")]
+        [InlineData("(this.Count) > (0)")]
+        [InlineData("((this).Count) > (0)")]
+        public Task CSharpTestFixOnParentheses(string condition)
+        {
+            string input = string.Format(CultureInfo.InvariantCulture, csSnippet, condition);
+            string fix = string.Format(CultureInfo.InvariantCulture, csSnippet, $"!{IsEmpty}");
+
+            return VerifyCS.VerifyCodeFixAsync(
+                 input,
+                 VerifyCS.Diagnostic(UseCountProperlyAnalyzer.s_rule_CA1836).WithSpan(10, 34, 10, 34 + condition.Length),
+                 fix);
+        }
+
+        [Theory]
+        [InlineData("(Count) > 0", "Not IsEmpty")]
+        [InlineData("Count > (0)", "Not IsEmpty")]
+        [InlineData("(Count) > (0)", "Not IsEmpty")]
+        [InlineData("(Me.Count) > 0", "Not IsEmpty")]
+        [InlineData("Me.Count > (0)", "Not IsEmpty")]
+        [InlineData("(Me.Count) > (0)", "Not IsEmpty")]
+        // TODO: Reduce suggested fix to avoid special casing here.
+        [InlineData("((Me).Count) > (0)", "Not (Me).IsEmpty")]
```

Comment:
For VB only, I had to manually specify the "expected fixed code" since depending on the parenthesis and on having the `Me` keyword, the result may chnage.

Why is this not a problem for C#? If you see above test `CSharpTestFixOnParentheses`, I always get the most "reduced" version when firing the rule.